src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2832,6 +2844,202 @@ ImmutableArray<IArgumentOperation> makeArguments(Lazy<ImmutableArray<IArgumentOp
             }
         }
 
+        public override IOperation VisitSwitch(ISwitchOperation operation, int? captureIdForResult)
+        {
+            Debug.Assert(_currentStatement == operation);
+
+            int expressionCaptureId = VisitAndCapture(operation.Value);
+
+            ImmutableArray<ILocalSymbol> locals = getLocals();
+            bool haveLocals = !locals.IsEmpty;
+            if (haveLocals)
+            {
+                EnterRegion(new RegionBuilder(ControlFlowGraph.RegionKind.Locals, locals: locals));
+            }
+
+            BasicBlock defaultBody = null; // Adjusted in handleSection
+            BasicBlock @break = GetLabeledOrNewBlock(operation.ExitLabel);
+
+            foreach (ISwitchCaseOperation section in operation.Cases)
+            {
+                handleSection(section);
+            }
+
+            if (defaultBody != null)
+            {
+                LinkBlocks(CurrentBasicBlock, defaultBody); 
+            }
+
+            if (haveLocals)
+            {
+                LeaveRegion();
+            }
+
+            AppendNewBlock(@break);
+
+            return null;
+
+            ImmutableArray<ILocalSymbol> getLocals()
+            {
+                ImmutableArray<ILocalSymbol> l = operation.Locals;
+                foreach (ISwitchCaseOperation section in operation.Cases)
+                {
+                    l = l.Concat(section.Locals);
+                }
+
+                return l;
+            }
+
+            void handleSection(ISwitchCaseOperation section)
+            {
+                var body = new BasicBlock(BasicBlockKind.Block);
+                var nextSection = new BasicBlock(BasicBlockKind.Block);
+
+                IOperation condition = ((BaseSwitchCase)section).Condition;
+                if (condition != null)
+                {
+                    Debug.Assert(!section.Clauses.Any(c => c.Label != null));
```

Comment:
>!section.Clauses.Any(c => c.Label != null) [](start = 33, length = 42)

This simplifies to `section.Clauses.All(c => c.Label == null)`, which is also significantly less mindbending to read and comprehend.